service/task.go

```diff
@@ -512,29 +512,24 @@ func (uis *UIServer) taskLog(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	defaultLogger, err := getDefaultLogger(projCtx)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	// check buildlogger logs first
+	var logReader io.ReadCloser
+	logReader, err = apimodels.GetBuildloggerLogs(r.Context(), uis.env.Settings().LoggerConfig.BuildloggerBaseURL, projCtx.Task.Id, logType, DefaultLogMessages, execution)
```

Comment:
Do you know how long this takes? I'm curious how much slower it will be to load logs if we wait for this to fail first.

Reply:
It didn't seem slow in staging, it should fail right away on the cedar side.

Reply:
SGTM, but will hold off on re-reviewing until the priority change.

Reply:
It actually checks the DB for the log before even doing authentication and fails right away if it DNE